Noting xref in module exception for release notes (#102280)
openshift/openshift-docs: contributing_to_docs/doc_guidelines.adoc

[IMPORTANT]
====

> Most modules cannot contain xrefs. If you _must_ use a link in your module, use the `link` attribute type and treat it as an external reference. The only exception is that xrefs are allowed in release notes modules, as long as they are only included one place.

Reviewer: ```suggestion
Most modules cannot contain xrefs. If you must include a link in a module, use the `link` attribute type and treat it as an external reference. The only exception is release notes modules, which may contain xrefs as long as the module is included in only one location.
```
